backend/dataall/modules/s3_datasets/aws/s3_dataset_client.py

```diff
@@ -73,3 +73,21 @@ def get_bucket_encryption(self) -> (str, str, str):
                     f'Data.all Environment Pivot Role does not have s3:GetEncryptionConfiguration Permission for {dataset.S3BucketName} bucket: {e}'
                 )
             raise Exception(f'Cannot fetch the bucket encryption configuration for {dataset.S3BucketName}: {e}')
+
+    def list_object_keys(self, bucket_name):
+        try:
+            response = self._client.list_objects_v2(
+                Bucket=bucket_name,
+            )
+
+            def txt_or_pdf(s):
+                suffix = s.split('.')[-1]
+                if suffix == 'pdf' or suffix == 'txt':
+                    return True
+                else:
+                    return False
+
```

Comment:
txt_or_pdf function is not needed:
`return [ob['Key'] for ob in response.get('Contents', []) if ob['Key'][ob['Key'].rfind('.')+1:] in ['pdf', 'txt']]`

Although for readability we might want to keep it as is

Reply:
it is essentially the same, right? I can make it a bit more clean